### PR TITLE
added base test

### PR DESCRIPTION
**Pull request by alexandra-protyanova**
Screenshot 2024-01-22 at 15 31 16

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1 +1,12 @@
...
+/* Don't change code below */
+
 'use strict';
+
+const { createServer } = require('./createServer');
+
+createServer()
+  .listen(5701, () => {
+    console.log(`Server is running on http://localhost:${5700} 🚀`);
+    console.log('Available at http://localhost:5700');
```
Comment (mgrinko) [inline src/app.js:11]: 5701 or 5700?

File: tests/paramsServer.test.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable max-len */
+'use strict';
```
Comment (mgrinko) [inline tests/paramsServer.test.js:2]: you could disable strict the same way

File: tests/paramsServer.test.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      let server;
+
+      beforeAll(() => {
+        server = createServer();
+        server.listen(PORT);
+      });
+
+      afterAll(() => {
+        server.close();
+      });
```
Comment (mgrinko) [inline tests/paramsServer.test.js:35]: Why do you do it after all but not each?
What if 1 test breaks the server 
The others will also fail

File: tests/paramsServer.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        expect(res.status).toEqual(200);
+      });
+
+      it('should handle pathnames with different lengths', async() => {
+        const response = await axios.get(HOST + '/a/b/c/d/e/f/g/h/i/j');
+
+        expect(response.data.parts).toEqual(['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']);
+
+        expect(response.status).toEqual(200);
+        expect(response.data.query).toEqual({});
```
Comment (mgrinko) [inline tests/paramsServer.test.js:66]: Why do we test the `query` here?

File: tests/paramsServer.test.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      });
+
+      it('should handle requests with no query parameters', async() => {
+        const response = await axios.get(HOST + '/test');
+
+        expect(response.data.query).toEqual({});
+        expect(response.data.parts).toEqual(['test']);
```
Comment (mgrinko) [inline tests/paramsServer.test.js:73]: Why do we test parts in the `query` test?

File: tests/paramsServer.test.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        expect(response.status).toEqual(200);
+      });
+
+      it('shouold handle requests with multiple query parameters', async() => {
+        const response = await axios.get(HOST + '/test?param1=value1&param2=value2');
+
+        expect(response.data.query).toEqual({
+          param1: 'value1', param2: 'value2',
```
Comment (mgrinko) [inline tests/paramsServer.test.js:81]:
```suggestion
          param1: 'value1',
          param2: 'value2',
```

File: tests/paramsServer.test.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        });
+      });
+
+      it('should not omit empty values for query params and pathname parts', async() => {
+        const response = await axios.post(HOST + '//test?param1=&param2=');
```
Comment (mgrinko) [inline tests/paramsServer.test.js:86]: do we really expect `//` to give us an `''`?

File: tests/paramsServer.test.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+      it('shouold handle requests with multiple query parameters', async() => {
+        const response = await axios.get(HOST + '/test?param1=value1&param2=value2');
+
+        expect(response.data.query).toEqual({
+          param1: 'value1', param2: 'value2',
+        });
+      });
+
+      it('should not omit empty values for query params and pathname parts', async() => {
```
Comment (mgrinko) [inline tests/paramsServer.test.js:85]: Every time you write a word `and` in the test you should think about splitting the test into parts

**Review by mgrinko — APPROVED** [review]
Looks good with some notes

File: src/app.js (hunk 1 above)
Comment (alexandra-protyanova) [reply src/app.js:11]: thanks, fixed

File: tests/paramsServer.test.js (hunk 2 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:2]: I don't see any benefits in disabling `use strict` rule for files, or you didn't mean it?

File: tests/paramsServer.test.js (hunk 3 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:35]: agree, fixed

File: tests/paramsServer.test.js (hunk 5 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:73]: looks like I shouldn't

File: tests/paramsServer.test.js (hunk 4 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:66]: fixed

File: tests/paramsServer.test.js (hunk 8 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:85]: fear

File: tests/paramsServer.test.js (hunk 7 above)
Comment (alexandra-protyanova) [reply tests/paramsServer.test.js:86]: it is not really clear from the condition of the task, more no than yes

**Pull request merged**